docs/HowToUsePeripheral.md

```diff
@@ -0,0 +1,101 @@
+# How to Use Peripheral
+
+Here are 4 parts, [UART](#uart), [SPI](#spi), [I2C](#i2c) and [I2S](#i2s) to use peripheral.
+
+## UART
+UART should provide two functions to initialize.  
+One is the **up_serialinit**, another is the **up_earlyserialinit**.
+
+### Add up_serialinit function
+This function registers character UART device drivers.
+```
+void up_serialinit(void)
+{
+	uart_register("dev/console", &CONSOLE_DEV);
+	uart_register("dev/ttyS0", &TTYS0_DEV);
+	uart_register("dev/ttyS1", &TTYS1_DEV);
+	...
+}
+```
+
+### Add up_earlyserialinit function
+Difference between this and above is calling location.  
+Os calls **up_serialinit** to initialize architecture specific functionality through *up_initialize*.  
+But boot code calls this before os starting. It performs the low level UART initialization early  
```

Comment:
As there is no explicit call to up_earlyserialinit() from OS,  bootup code must call it prior to os start in order to provide low level debug message.

Hence it would be better to change the statement to "boot code  *must* call this before os starting, to provide low level debug message.

Reply:
You mean you want to add "must" in sentence. Right?

Reply:
yes